### DISCUSSION TITLE
Custom helper each with comma

### DISCUSSION THREAD

**Discussion by jcbenck** [mockoon/mockoon#1239] [Help] [answered]
I am currently working on formatting a JSON array using Mockoon's templating system, and I've encountered a challenge with the each helper, specifically regarding the placement of commas.

When I use the repeat helper, it correctly adds a comma after each object in the array and omits the comma for the last object, as expected. Here's an example:
```
[
  {{# repeat 50 }}
  {
    "id": "{{faker 'string.uuid'}}",
    "name": "{{faker 'company.name'}}"
  }
  {{/ repeat }}
]
```
However, I'm trying to achieve a similar result with the each helper. When I use the each helper like this:
```
[
  {{#each (dataRaw 'abcd')}}
    {
      "name": "{{this.name}}"
    }
  {{/each}}
]
```
I don't get a comma after each object. But if I try to manually add a comma:
```
[
  {{#each (dataRaw 'abcd')}}
    {
      "name": "{{this.name}}"
    },
  {{/each}}
]
```
It results in an extra comma after the last object, making the JSON invalid. Is there a way in Mockoon's templating system to correctly place a comma after each object except the last, similar to how it works in the repeat helper? Any advice or example syntax for this would be greatly appreciated.

Thank you for your help!

**Comment by 255kb** [comment, marked as answer]
The `repeat` helper is a custom one we create that automatically add the comma. The `each` is the one from Handlebars that is only iterating. To add a comma except on the last item, you can use `unless` ([also from Handlebars](https://handlebarsjs.com/guide/builtin-helpers.html#unless)):

```
[
  {{#each (dataRaw 'abcd')}}
    {
      "name": "{{this.name}}"
    }{{#unless @last}},{{/unless}}
  {{/each}}
]
```
